aqt/archives.py

```diff
@@ -442,11 +442,12 @@ def _get_archives_base(self, name, target_packages):
                 try:
                     extensions_xml_text = self._download_update_xml(extensions_xml_url, True)
                 except ArchiveDownloadError:
-                    # In case _download_update_xml ignores the hash and tries to get the url.
+                    # In case _download_update_xml failed to get the url because of no extension.
                     pass
-                if extensions_xml_text:
-                    self.logger.info("Found extension {}".format(ext))
-                    update_xmls.append(UpdateXmls(extensions_target_folder, extensions_xml_text))
+                else:
+                    if extensions_xml_text:
+                        self.logger.info("Found extension {}".format(ext))
+                        update_xmls.append(UpdateXmls(extensions_target_folder, extensions_xml_text))
```

Comment:
I moved this into the try block but putting them in the else block may be preferred.  There are two places in metadata.py
related to "except (ChecksumDownloadFailure, ArchiveDownloadError)" where I also added additional code to the try block where an else block might be preferred.